Release 6.13.0

ruyadorno opened the pull request:
## 6.13.0 (2019-11-05)

* [`9a2d8af84`](https://github.com/npm/cli/commit/9a2d8af84f7328f13d8f578cf4b150b9d5f09517) [#240](https://github.com/npm/cli/pull/240) Clean up some test flakiness and inconsistency ([@isaacs](https://github.com/isaacs))
* [`688cd97be`](https://github.com/npm/cli/commit/688cd97be94ca949719424ff69ff515a68c5caba) [#272](https://github.com/npm/cli/pull/272) use github actions for CI ([@JasonEtco](https://github.com/JasonEtco))

### DEPENDENCIES

* [`a37296b20`](https://github.com/npm/cli/commit/a37296b20ca3e19c2bbfa78fedcfe695e03fda69) `pacote@9.5.9`

isaacs committed:
test: Clean up some flakiness and inconsistency

* Get rid of a lot of usage of osenv.tmpdir in tests.
* Remove unnecessary creation/deletion of common.pkg
* Reduce rimraf.sync wherever possible (often collides with
  common-tap.js's rimraf on windows)
* Use common test utilities wherever possible.
* DRY tests with test templates where it makes sense.

PR-URL: https://github.com/npm/cli/pull/240
Credit: @isaacs
Close: #240
Reviewed-by: @ruyadorno

kemitchell committed:
support: add support subcommand

PR-URL: https://github.com/npm/cli/pull/246
Credit: @kemitchell
Close: #246
Reviewed-by: @ruyadorno

Thanks @kemitchell for providing the initial work that served as a base
for `npm fund`, its original commits messages are preserved as such:

- support: add support subcommand
- support: fix request caching
- support: further sanitize contributor data
- doc: Fix typo
- support: simplify to just collecting and showing URLs
- install: improve `npm support` test
- install: drop "the" before "projects you depend on"
- doc: Reword mention of `npm support` in `package.json` spec

claudiahdz committed:
docs: migrate existing docs to gatsby

- chore(docs): remove html from docs build
- feat(docs): migrate new Gatsby site to cli/docs
- chore(docs): fix docs url path
- chore(docs): add new build doc step
- chore(docs): add static linking
- chore(docs): add npm favicon
- chore(docs): update files for tests
- chore(docs): make relative paths absolute
- chore(docs): fix routing on docs dependant tests
- chore(docs): add prepublishOnly docs building step
- chore(docs): add docs/public to .gitignore
- chore(docs): rename dot json markdown files with hyphen names
- chore(docs): clean package.json
- chore(docs): highlight scripts in list
- chore(docs): add strong styles
- chore(docs): fix .json links on markdown
- Added copy to the features page
  - Added an install page (copied from existing online docs)
  - Added a FeatureLink component to components/links.js
- feat(docs): replace docs sh script with node
- chore(docs): fix docs building process
- chore(docs): docs folder cleanup

PR-URL: https://github.com/npm/cli/pull/274
Credit: @claudiahdz
Close: #274
Reviewed-by: @ruyadorno

ruyadorno committed:
fund: add fund command

This commit introduces the `npm fund` command that lists all `funding`
info provided by the installed dependencies of a given project.

Notes on implementation:

- `lib/utils/funding.js` Provides helpers to validate funding info and
return a tree-shaped structure containing the funding data for all deps.
- `lib/fund.js` Implements `npm fund <pkg>` command
- Added tests
  - `npm install` mention of funding
  - `npm fund <pkg>` variations
  - unit tests for added `lib/utils` and `lib/install` helpers
- Added docs for `npm fund`, `funding` `package.json` property
- Fixed `lib/utils/open-url` to support `--json` config
- Documented `unicode` on `npm install` docs
- fix tests
- fix planned tap tests
- alternative solution to --no-browser arg
- docs: moved fund docs to new location

Refs: https://github.com/npm/rfcs/blob/2d2f00457ab19b3003eb6ac5ab3d250259fd5a81/accepted/0017-add-funding-support.md

PR-URL: https://github.com/npm/cli/pull/273
Credit: @ruyadorno
Close: #273
Reviewed-by: @darcyclarke

JasonEtco committed:
use github actions for CI

PR-URL: https://github.com/npm/cli/pull/272
Credit: @JasonEtco
Close: #272
Reviewed-by: @ruyadorno

oletizi committed:
docs: added copy to the features page

- Added an install page (copied from existing online docs)
- Added a FeatureLink component to components/links.js

ruyadorno committed:
pacote@9.5.9

ruyadorno committed:
docs: CHANGELOG updates for 6.13.0

oletizi committed:
docs: Updated lorem ipsum with real copy.

Modified the copy of the "Publish" card to make the transition to
npmjs.com/products less jarring

PR-URL: https://github.com/npm/cli/pull/277
Credit: @oletizi
Close: #277
Reviewed-by: @ruyadorno

ljharb committed:
unsupported: npm v6.10.0 breaks on node v6.0, v6.1, v9.0 - v9.2

https://github.com/nvm-sh/nvm/commit/100861d529d26a4022ee4ac6dc5d8df86a3a6f65

PR-URL: https://github.com/npm/cli/pull/279
Credit: @ljharb
Close: #279
Reviewed-by: @ruyadorno

Commit:
fix: on uninstall package, *.ps1 not delete

PR-URL: https://github.com/npm/cli/pull/281
Credit: @NoDocCat
Close: #281
Reviewed-by: @ruyadorno

cclauss commented:
Will node-gyp v6.0.1 be part of this release?  It contains some important compatibility fixes for Python 3.  https://github.com/nodejs/node-gyp/blob/master/CHANGELOG.md

ljharb commented:
@ruyadorno itd be great to get #279 into this version as well.

darcyclarke committed:
chore: create settings.yml file

- This will ensure we inherit our standard labels from the `open-source-project-boilerplate`

PR-URL: https://github.com/npm/cli/pull/282
Credit: @darcyclarke
Close: #282
Reviewed-by: @ruyadorno

ruyadorno committed:
docs: removed refs to npm-index

isaacs committed:
doc: Move @VERSION@ for html docs into gatsby

isaacs committed:
doc: fix hyperlink refs in man page docs

PR-URL: https://github.com/npm/cli/pull/283
Credit: @isaacs
Close: #283
Reviewed-by: @ruyadorno

ruyadorno committed:
read-cmd-shim@1.0.5

ruyadorno committed:
docs: updated changelog for 6.13.0

ruyadorno committed:
6.13.0

ruyadorno committed:
update AUTHORS
